test/gurka/test_turn_lanes.cc

```diff
@@ -393,6 +393,145 @@ TEST(Standalone, TurnLanesUTurns) {
+       {"FL",
+        {{"highway", "primary"},
+         {"oneway", "yes"},
+         {"name", "Patuxent Woods Drive"},
+         {"lanes", "3"},
+         {"turn:lanes", "||right"}}},
+       {"LM",
+        {{"highway", "primary"},
+         {"oneway", "yes"},
+         {"name", "Patuxent Woods Drive"},
+         {"lanes", "3"},
+         {"turn:lanes", "||right"}}},
+       {"MQ", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Patuxent Woods Drive"}}},
+       {"RO", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Patuxent Woods Drive"}}},
+       {"OI", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Patuxent Woods Drive"}}},
+       {"IG", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Patuxent Woods Drive"}}},
+       {"LI", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Small Internal Street"}}},
+       {"MO", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Small Internal Street"}}},
+       {"LP", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "First Street"}}},
+       {"MN", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Second Street"}}},
+       {"GB", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+       {"BJ", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+       {"KC", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+       {"CF", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}}};
+
+  const gurka::relations relations = {
+      {{
+           {gurka::way_member, "CF", "from"},
+           {gurka::way_member, "GB", "to"},
+           {gurka::way_member, "FG", "via"},
+       },
+       {
+           {"type", "restriction"},
+           {"restriction", "no_u_turn"},
+       }},
+      {{
+           {gurka::way_member, "FL", "from"},
+           {gurka::way_member, "IG", "to"},
+           {gurka::way_member, "LI", "via"},
+       },
+       {
+           {"type", "restriction"},
+           {"restriction", "no_u_turn"},
+       }},
+  };
+
+  const auto layout =
+      gurka::detail::map_to_coordinates(ascii_map, gridsize_metres, {5.1079374, 52.0887174});
+
+  auto map = gurka::buildtiles(layout, ways, {}, relations, "test/data/gurka_turn_lanes_7");
+
+  valhalla::Api result;
+
+  // Test U-turn using left-only lanes. The left-most left lane should be
+  // active & rest invalid
+  result = gurka::route(map, "A", "H", "auto");
+
+  gurka::assert::raw::expect_maneuvers(result, {DirectionsLeg_Maneuver_Type_kStart,
+                                                DirectionsLeg_Maneuver_Type_kUturnLeft,
+                                                DirectionsLeg_Maneuver_Type_kDestination});
+  validate_turn_lanes(result, {
+                                  {5, "[ *left* ACTIVE | left | through | right | right ]"},
+                                  {5, "[ left | left | *through* ACTIVE | right | right ]"},
```

Comment:
I think this is a problem with the recent update to change the maneuver placement on internal intersections. When the turn lanes are being processed and activated (`ProcessTurnLanes`) the internal edge is part of the next maneuver, and we only update the placement of the maneuver after the turn lanes are processed. Since the internal edge already has turn lanes we don't update them, and so we're left with the wrong turn lane being activated.

@dgearhart do you have any thoughts on how we could address this? One idea I had was to move `ProcessTurnLanes` after `MoveInternalEdgeToPreviousManeuver`.

Reply:
Yeah, thats probably whats happening. I think always copying the turn lane info of the previous edge for internal intersections seems like a good solution (it would avoid the situation with the left-uturn example below too)

Reply:
@dgearhart this is the issue I am seeing a lot now.

Reply:
@gknisely @danpaz @mandeepsandhu agreed - we should try to move `ProcessTurnLanes` after `UpdateManeuverPlacementForInternalIntersectionTurns` and retest